git_credentials/credentials.py

```diff
@@ -25,11 +25,12 @@ class GitCredentialDescription(NamedTuple):
 
 class GitCredentials:
     """A class to wrap around git-credentials. See it's man page for more info."""
-    def __init__(self, repository_path: str, env: Optional[Dict[str, str]] = None):
+    def __init__(self, repository_path: Optional[str]=None, env: Optional[Dict[str, str]] = None):
         """Creates a new environment for git-credential commands to run in."""
         self.repository_path = repository_path
         self.env = env if env else os.environ.copy()
         self.env['GIT_TERMINAL_PROMPT'] = '0'
+        self.env['GIT_ASKPASS']='0'
```

Comment:
This might not always be a good idea, I just this in a project where falling back to asking for the PW is actually behavior I need. Can you put this behind a flag to the constructor?

Reply:
Yes of course, I have added the flag.

Reply:
Being unsure about docstring format, I am giving you edits rights to the PR